examples/dashboards/dashboards.md

```diff
@@ -10,7 +10,7 @@ There exists Grafana dashboards for each component (not all of them complete) ta
 - [Thanos Receiver](receive.json)
 - [Thanos Sidecar](sidecar.json)
 - [Thanos Ruler](rule.json)
-- [Thanos Replicate](bucket_replicate.json)
+- [Thanos Replicate](bucketReplicate.json)
```

Comment:
```suggestion
- [Thanos Replicate](bucket_replicate.json)
```

I think we should keep this convention and change the `queryFrontend.json` above for filenames.

Reply:
I tried to do this [here](https://github.com/thanos-io/thanos/pull/4856#discussion_r747791376) but it was failing, because we were using the first part of the filename to set the `component` variable: https://github.com/thanos-io/thanos/blob/main/mixin/dashboards/defaults.libsonnet#L16

So for this case, we would get `component = bucket_replicate` from `bucket_replicate.json`, instead of `bucketReplicate`.

I thought then we could have a jsonnet function that handles this case - so in case we can split the filename by `_`, we update the component name to use `camelCase` instead of `snake_case`.

I tried to do this in this commit: https://github.com/thanos-io/thanos/pull/4859/commits/aa46b6b5fc9bfba6c2f7b34a145dc8a06b78af5d so now it should hopefully work, either if we use filenames as `bucket_replicate.json`, `query_frontend.json`  or if we use 'simple filenames' - e.g. `query.json`

I'm new to jsonnet so maybe the code is not that readable/there are better ways to do this, but the idea was to split the component name by `_` and, in case we have more then 1 word, then we make the first letter of the other words uppercase, then we join all of them in one word again to have the component name in camelCase format.

Let me know what you think.. I can also move this commit/the related changes to other PR :)

Reply:
This looks good to me. I think you did an amazing job :+1: